plugins/modem-manager/fu-mm-device.c

```diff
@@ -1504,6 +1546,35 @@ fu_mm_device_attach(FuDevice *device, FuProgress *progress, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_mm_device_setup(FuDevice *device, GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE(device);
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(GError) error_branch = NULL;
+
+	/* Create IO channel to send AT commands to the modem */
+	locker = fu_device_locker_new_full(device,
+					   (FuDeviceLockerFunc)fu_mm_device_io_open,
+					   (FuDeviceLockerFunc)fu_mm_device_io_close,
+					   &error);
+	/*
+	 * firmware branch AT command may fail if not implemented,
+	 * clear error if not supported
+	 */
+	if (self->branch_at != NULL) {
```

Comment:
tiny nit: error_branch could be created here

Reply:
Ah don't we declare all variables at the beginning of a function?

Reply:
in general we try to err on creating them as "local" as possible, but sticking them at the top of whatever block.